### PR TITLE
Single node functionality

### PR DESCRIPTION
**Pull request by mattkasun**

### CODE REVIEW

File: config/node.go [hunk 1]
```diff
@@ -91,7 +105,7 @@ func (node *Node) PrimaryAddress() net.IPNet {
 	return node.Address6
 }
 
-// writeNodeConfiguation writes the node map to disk
+// WriteNodeConfiguation writes the node map to disk
```
Comment (0xdcarns) [inline config/node.go:108]: comment does not match func name

File: functions/list.go [hunk 2]
```diff
@@ -1,18 +1,9 @@
...
 
-	"github.com/devilcove/httpclient"
 	"github.com/gravitl/netclient/config"
-	"github.com/gravitl/netmaker/logger"
-	"github.com/gravitl/netmaker/models"
-	"github.com/kr/pretty"
-	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
 )
 
 type Peer struct {
```
Comment (0xdcarns) [inline functions/list.go:9]: this and CIDR struct below are missing comments

File: functions/mqhandlers.go [hunk 3]
```diff
@@ -92,41 +85,23 @@ func NodeUpdate(client mqtt.Client, msg mqtt.Message) {
...
-		if err != nil {
-			logger.Log(0, "network:", newNode.Network, "error modifying node port on", newNode.Name, "-", err.Error())
-			return
-		}
-		ifaceDelta = true
-		informPortChange(newNode)
-	}
-	if err := wireguard.UpdateWgInterface(file, nameserver, newNode); err != nil {
+	nc := wireguard.NewNCIface(config.Netclient(), config.GetNodes())
+	nc.Create()
```
Comment (0xdcarns) [inline functions/mqhandlers.go:93]: these should have error checks, maybe it isn't necessary right now, but in future

File: wireguard/wireguard_conf.go [hunk 4]
```diff
@@ -151,85 +156,100 @@ func UpdatePrivateKey(file, privateKey string) error {
...
 		}
 	}
-	if err := wireguard.SaveTo(config.GetNetclientInterfacePath() + node.Interface + ".conf"); err != nil {
+	if err := wireguard.SaveTo(config.GetNetclientPath() + host.Interface + ".conf"); err != nil {
 		return err
 	}
 	return nil
 }
+
+func AddAddresses(node *config.Node) {
```
Comment (0xdcarns) [inline wireguard/wireguard_conf.go:239]: missing comment

File: wireguard/wireguard_linux.go [hunk 5]
```diff
@@ -111,7 +116,7 @@ func (nc *NCIface) ApplyAddrs() error {
 type netLink struct {
 	attrs *netlink.LinkAttrs
 }
-
+pretty
```
Comment (0xdcarns) [inline wireguard/wireguard_linux.go:119]: random word here

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: functions/list.go (hunk 2 above)
Comment (mattkasun) [reply functions/list.go:9]: structs were redundant, removed

File: config/config.go [hunk 6]
```diff
@@ -30,66 +31,79 @@ const (
...
+	InternetGateway   net.UDPAddr
 }
 
 func init() {
 	Servers = make(map[string]Server)
 	Nodes = make(map[string]Node)
 }
 
+// UpdateNetcllient updates the in menory version of the host configuration
+func UpdateNetclient(c Config) {
```
Comment (abhishek9686) [inline config/config.go:72]: typo in func comment  `UpdateNetcllient` --> `UpdateNetclient`

File: functions/mqhandlers.go (hunk 3 above)
Comment (mattkasun) [reply functions/mqhandlers.go:93]: added error handling

File: functions/join.go [hunk 7]
```diff
@@ -265,51 +246,29 @@ func JoinViaSSo(flags *viper.Viper) (*models.AccessToken, error) {
 	}
 }
 
-// JoinNetwork - helps a client join a network
-func JoinNetwork(flags *viper.Viper) (*config.Node, *config.Server, error) {
-	netclient := &config.Netclient
+// JoinNetwork - connects to netamker server to join a network
```
Comment (abhishek9686) [inline functions/join.go:249]: typo in the comment  `netamker` --> `netmaker`

**Pull request merged**